examples/gno.land/p/demo/authctx/authctx.gno

```diff
@@ -0,0 +1,162 @@
+// Package authctx offers a versatile framework for handling authentication contexts.
+// It empowers the management of authentication contexts for users, realms, delegated calls,
+// and potentially extends its capabilities to non-address entities and more.
+package authctx
+
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+)
+
+// AuthContext is an interface that represents an authentication context.
+type AuthContext interface {
+	Addr() std.Address
+	String() string
```

Comment:
Can we document what you wrote earlier about the requirements for these?

Reply:
```suggestion
	// String returns a unique and human-readable identifier, typically in the format "<type>:<addr>[/opts]".
	String() string
```

Reply:
`AuthContext` represents an owner, either direct (Orig/Prev) or indirect (Delegated). It's not a comprehensive ACL tool—this might be better suited for a separate library.

To initiate an `AuthContext`, you must be the original owner, achievable through the `Orig/Prev` constructors or by `Approving` another realm/account for future delegated authentication. We might introduce `authctx.IsNative(ctx)` to determine the origin of the `AuthContext` implementation.

`IsAuthorized` defines scope, not full access, mirroring JWTs. We opt for Golang's private fields for authenticity verification. Maybe HasScope() would be a better name than IsAuthorized.

A feature in `AuthContext` allows privilege adjustments. This explains why `IsAuthorized` defaults to true for direct contexts but needs a specified scope for upcoming delegations. Then you can re-Scope (drop privileges) on any AuthContext.

In essence, `AuthContext` isn't purely for ACLs. Manage ACLs with tools like `p/acl`. For delegated contexts, scope is as important as ACLs. While ACLs verify an owner's actions, `AuthContext.IsAuthorized` assesses the caller's rights to claim ownership in the present scenario.

Regarding potential paths forward (that I would like to experiment in the upcoming days): 
1. Removing `IsAuthorized`, making `AuthContext` a container, and placing checks during object creation.
2. Rename `IsAuthorized` with `HasScope`.
3. Enabling custom `IsAuthorized` checks through closures, allowing diverse filtering methods.
4. Refining the current `p/acl` library, making it interface-based, and employing that interface for `IsAuthorized`.